### PR TITLE
[Engine] Apply the STS task to bge models

### PR DESCRIPTION
**Pull request by Zhenzhong1**
## Type of Change

* Example add 
* API extend
* Doc update

## Description

* Add the new directory for the text-embedding task and Massive Text Embedding Benchmark (mteb).
* Verifyed the [BAAI/bge-small-en-v1.5](https://huggingface.co/BAAI/bge-large-en-v1.5) accuracy with the Neural Engine backend.
* Verifyed the [BAAI/bge-base-en-v1.5](https://huggingface.co/BAAI/bge-large-en-v1.5) accuracy with the Neural Engine backend.

Accuracy
int8 bge-base with the Neural Engine on the STS: **82.191**
image

int8 bge-small with the Neural Engine on the STS: **81.43**
image

Usage:
```python

from transformers import AutoTokenizer
from intel_extension_for_transformers.transformers import AutoModel

sentences_batch = ['sentence-1', 'sentence-2', 'sentence-3', 'sentence-4']

tokenizer = AutoTokenizer.from_pretrained('BAAI/bge-base-en-v1.5')
encoded_input = tokenizer(sentences_batch,
                            padding=True,
                            truncation=True,
                            max_length=512,
                            return_tensors="np")

engine_input = [encoded_input['input_ids'], encoded_input['token_type_ids'], encoded_input['attention_mask']]

model = AutoModel.from_pretrained('./model_and_tokenizer/int8-model.onnx', use_embedding_runtime=True)
sentence_embeddings = model.generate(engine_input)['last_hidden_state:0']

print("Sentence embeddings:", sentence_embeddings)
```

## Expected Behavior & Potential Risk

N/A

## How has this PR been tested?

[bge benchmark](https://intel-my.sharepoint.com/:x:/p/yuwen_zhou/EZyv9bXh2oxMsLFjFrxzCbcB7EJQScJamOOHbz4rhjqnqA?email=zhenzhong.xu%40intel.com&e=4%3AvqVeiB&fromShare=true&at=9)

## Dependency Change?

remove C_MTEB

### CODE REVIEW

**Review by a32543254 — APPROVED** [review]
LGTM

**Comment by a32543254** [comment]
do we need some extension test to make sure the func of bge is all right?

**Review by yuwenzho — APPROVED** [review]

**Comment by VincyZhang** [comment]
Will test LLM deprecated python mode in release tests.
